app/screens/ChatbotScreen.tsx

```diff
@@ -0,0 +1,361 @@
+        let recreatedAnswer = newPages[newPages.length-1].answer
+        if (recreatedAnswer !== undefined) {
+          requestNextStep(chatbotCreateAnswer(newValue, recreatedAnswer), recreatedAnswer, chatbotId, newPages)
+        }
+      })
+  }
+
+  const requestNextStep = (apiAnswer: ChatbotAPIAnswer, answer?: ChatbotAnswer, chatbotId?: string, inputPages?: ChatbotPage[]): Promise<ChatbotPage[]> => {
+    setRequestingNextStep(true)
+    if (inputPages === undefined) {
+      inputPages = pages
+    }
+    return nextStep(sessionId, chatbotId, apiAnswer)
+      .then((question) => {
+        setRequestingNextStep(false)
+        if (answer !== undefined) {
+          chatbotFillAnswerWithData(question, answer)
+        }
+        let [newPages, isFinished, help] = chatbotFeedQuestion(question, settings?.language)
+        let combinedPages: ChatbotPage[] = []
+        if (help !== undefined) {
+          NotificationService.error(help)
+        } else {
+          if (newPages.length > 0) {
+            let combinedPages = inputPages?.concat(newPages) ?? newPages
+            setPages(combinedPages)
+            onNext(combinedPages, isFinished, answer)
+          } else {
+            NotificationService.error(t("kyc.bot.error.validation"))
+          }
+        }
+        setLoading(false)
+        return combinedPages
+      })
+  }
+
+  const updateProgress = (pages: ChatbotPage[], pageIndex: number, isBack: boolean = false) => {
+    if (pageIndex <= 0) {
+      setProgress(0)
+    }
+    // Krysh: there are always around 20 questions. might be more in some cases.
+    // current page / max pages
+    let newProgress = Math.max(pageIndex, 1) / Math.max(pages.length, 20)
+    // on back allow lower progress to previous one
+    if (isBack) {
+      setProgress(newProgress)
+    } 
+    // otherwise only higher progress are allowed, to avoid any confusion for users
+    else if (newProgress > progress) {
+      setProgress(newProgress)
+    }
+  }
+
+  return (
+    <View style={styles.container}>
+      {/* Loading */}
+      {isLoading && <Loading size="large" />}
+      {/* SMS screen */}
+      {!isLoading && !isSMSCompleted && (
+        <View style={styles.container}>
+          <View>
+            <H2 text={t("kyc.bot.sms_header")} />
+            <SpacerV />
+            <View style={[AppStyles.containerHorizontal, styles.header]}>
+              <H3 text={t("kyc.bot.sms_code")} />
+              <IconButton color={Colors.Primary} icon="reload" onPress={() => { requestSMS(sessionId) }} />
+            </View>
+            <SpacerV />
+            <TextInput value={smsCode} onChangeText={setSMSCode} onSubmitEditing={() => { submitSMSCode(sessionId) }} placeholder={t("kyc.bot.sms_placeholder")} keyboardType="numeric" />
+            <SpacerV />
+            <Text style={{ color: Colors.Grey }}>{t("kyc.bot.sms_help")}</Text>
+            <TouchableHighlight onPress={() => Linking.openURL('mailto:support@kyc.ch')}>
```

Comment:
better use TouchableOpacity for a unified experience throughout the app
(+ https://stackoverflow.com/a/46075892/1176551 & https://stackoverflow.com/a/42406240/1176551